sled-agent/src/bootstrap/http_entrypoints.rs

```diff
@@ -66,3 +68,29 @@ async fn rack_initialize(
     ba.rack_initialize(request).await.map_err(|e| Error::from(e))?;
     Ok(HttpResponseUpdatedNoContent())
 }
+
+/// Resets the rack to an unconfigured state.
+#[endpoint {
+    method = DELETE,
+    path = "/rack-initialize",
+}]
+async fn rack_reset(
+    rqctx: RequestContext<Arc<Agent>>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let ba = rqctx.context();
+    ba.rack_reset().await.map_err(|e| Error::from(e))?;
```

Comment:
Apologies for the vagueness of this question - do we need any kind of locking / queuing to handle this endpoint being hit multiple times before the first request finishes? Similarly for `sled_reset` and/or interactions between reset and initialize.

Reply:
The `sled_reset` function accesses the `sled_state` variable of the bootstrap agent, which is protected by a `Mutex`. This means, at a baseline, all accesses to `sled_reset` are serialized.

I *do* think it's technically possible to POST and DELETE `rack-initialize` concurrently. I can add a mutex to ensure that only one of those methods can be called at once.

This is done in https://github.com/oxidecomputer/omicron/pull/2677/commits/3fc2c7ba43ee0d0a3ffa3d40d45ae27b390d9ae8